### PR TITLE
Add options to remove

### PR DESCRIPTION
**Pull request by ryansch**

### CODE REVIEW

File: lib/tire/index.rb [hunk 1]
```diff
@@ -335,7 +335,11 @@ def remove(*args)
       end
       raise ArgumentError, "Please pass a document ID" unless id
 
-      url    = "#{self.url}/#{type}/#{Utils.escape(id)}"
+      options ||= {}
+      options.select! { |k, | [:routing, :parent, :replication, :consistency, :refresh, :timeout].include? k }
```
Comment (morganick) [inline lib/tire/index.rb:339]: Do we need to add all of these options? I understand `:routing` but why the rest?

File: lib/tire/index.rb [hunk 2]
```diff
@@ -335,7 +335,11 @@ def remove(*args)
...
+      params_encoded = options.empty? ? '' : "?#{options.to_param}"
+
+      url    = "#{self.url}/#{type}/#{Utils.escape(id)}#{params_encoded}"
```
Comment (morganick) [inline lib/tire/index.rb:342]: `Utils.escape(type)`

**Review by morganick — CHANGES_REQUESTED** [review]

File: lib/tire/index.rb (hunk 1 above)
Comment (ryansch) [reply lib/tire/index.rb:339]: Those are the options that the endpoint can accept via the 0.90 documentation.

File: lib/tire/index.rb (hunk 2 above)
Comment (ryansch) [reply lib/tire/index.rb:342]: This is how every method in tire is currently written.

Comment (morganick) [reply lib/tire/index.rb:342]: Yes and, I found where it was escaped though:
https://github.com/outstand/tire/blob/18942917a07ed4ea278f7b3686f30dc187608d1b/lib/tire/index.rb#L329

I'm assuming that getting type from document is already escaped?
https://github.com/outstand/tire/blob/18942917a07ed4ea278f7b3686f30dc187608d1b/lib/tire/index.rb#L333

This is the line that made me think about it:
https://github.com/outstand/tire/blob/18942917a07ed4ea278f7b3686f30dc187608d1b/lib/tire/index.rb#L354

Comment (ryansch) [reply lib/tire/index.rb:342]: Fixed.

File: lib/tire/index.rb [hunk 3]
```diff
@@ -339,7 +339,7 @@ def remove(*args)
       options.select! { |k, | [:routing, :parent, :replication, :consistency, :refresh, :timeout].include? k }
       params_encoded = options.empty? ? '' : "?#{options.to_param}"
 
-      url    = "#{self.url}/#{type}/#{Utils.escape(id)}#{params_encoded}"
+      url    = "#{self.url}/#{Utils.escape(type)}/#{Utils.escape(id)}#{params_encoded}"
```
Comment (morganick) [inline lib/tire/index.rb]: Is there any issue with escaping it twice? (because of line 239?)

Comment (ryansch) [reply lib/tire/index.rb]: 239 is a blank line

Comment (ryansch) [reply lib/tire/index.rb]: 329.

You're right now we're double escaping.

File: lib/tire/index.rb (hunk 2 above)
Comment (ryansch) [reply lib/tire/index.rb:342]: Unfixed!

**Review by morganick — APPROVED** [review]

**Pull request merged**